File: backend/models/chess_board.py
from pydantic import BaseModel
from typing import Optional, List
from models.figure import Figure

class ChessBoard(BaseModel):
    
    squares: List[List[Optional[Figure]]]

    @classmethod
    def create_empty_board(cls):
        return cls(squares=[[None for _ in range(8)] for _ in range(8)])

File: frontend/src/models/ChessBoard.ts
import { Figure } from "./Figure";

export class ChessBoard {
    squares: (Figure | null)[][];

    constructor() {
        this.squares = Array(8).fill(null).map(() => Array(8).fill(null));
    }

    static createEmptyBoard(): ChessBoard {
        return new ChessBoard();
    }
}
